### ISSUE TITLE
Citation generator

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/MyopicVicar#1374]
This feature would auto-generate a citation that could be cut and pasted by researchers. 
 Probably needs a variety of forms, to reflect requirements of different online / collaborative works and academic publication standards - although the latter could be via http://www.citationmachine.net/ or similar. 
See https://fhiso.org/TR/cev-concepts for draft standard
Dependent on #801 (persistent urls) and coordination board #75 (citations for Free UK Gen databases)

Priority: 3,3,5,4: 15
Depends on #1695
Needs #1772
Needs Issue#1098

**Comment by raghavbhatnagar96** [comment]
Hi. I am Raghav from IIIT Bangalore, India. I would like to work on this as a part of GSoC 2018. I have filled the google form but have not received a slack invite. I am deploying the application but need the database.yml file (according to https://docs.google.com/document/d/11n5F9WB9WA9BgZwj1QDJf2OdZOPO1-jkdY1cXOU-AHE/edit?usp=sharing). I have installed all prerequisites and this is the next step that requires a mentor to send the file as it is not there in the repository. I have worked on Ruby on rails and MySQL among other languages. I would like to get in touch with a mentor as soon as possible so that I can discuss the project with him/her.

**Comment by PatReynolds** [comment]
Examples of genealogy citation: 
https://www.wikitree.com/wiki/Help:Sources
https://tienle47.wordpress.com/tag/genealogy-citations/
https://www.thoughtco.com/how-to-cite-genealogy-sources-1421785

**Comment by PatReynolds** [comment]
Contacted Wikitree, Family Tree Maker and Evidence Explained 13/9/2018

**Comment by richpomfret** [comment]
@sudaraka94 @benwbrum can you provide an update on any final/remaining subtasks here that need to be completed or has this now been finished? :)

**Comment by Captainkirkdawson** [comment]
The current citation to the website is just to the main site there is no specific record url

**Comment by richpomfret** [comment]
@sudaraka94 do you know have everything you need? Would be great to have this finished/deployed over the next sprint. Please work from the current master.

**Comment by richpomfret** [comment]
@benwbrum to merge the code so we can test on test3.

**Comment by Captainkirkdawson** [comment]
The citation code is already running and available on test3 just needs updating it there are some changes

On February 20, 2019, at 7:14 PM, Rich Pomfret <notifications@github.com> wrote:

@benwbrum to merge the code so we can test on test3.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.￼

{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/FreeUKGen/MyopicVicar","title":"FreeUKGen/MyopicVicar","subtitle":"GitHub repository","main_image_url":"<a href="https://github.githubassets.com/images/email/message_cards/header.png">https://github.githubassets.com/images/email/message_cards/header.png</a>","avatar_image_url":"<a href="https://github.githubassets.com/images/email/message_cards/avatar.png">https://github.githubassets.com/images/email/message_cards/avatar.png</a>","action":{"name":"Open in GitHub","url":"<a href="https://github.com/FreeUKGen/MyopicVicar">https://github.com/FreeUKGen/MyopicVicar</a>"}},"updates":{"snippets":[{"icon":"PERSON","message":"@richpomfret in #1374: @benwbrum to merge the code so we can test on test3."}],"action":{"name":"View Issue","url":"<a href="https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096">https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096</a>"}}} [ { "@context": "<a href="http://schema.org">http://schema.org</a>", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "<a href="https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096">https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096</a>", "url": "<a href="https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096">https://github.com/FreeUKGen/MyopicVicar/issues/1374#issuecomment-465671096</a>", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "<a href="https://github.com">https://github.com</a>" } } ]

**Comment by richpomfret** [comment]
@benwbrum Are any further updates needed for this? I think Sudaraka completed his changes, however I wonder if they are now on TEST3 or not. If no further changes are needed, please move to need review or ready for test - I've had a look on TEST3 and it looks fine to me, but @PatReynolds will need to sign off on final testing.

**Comment by Captainkirkdawson** [comment]
Propose to integrate the current citation code with the code resulting from #1854

**Comment by Captainkirkdawson** [comment]
@PatReynolds @richpomfret @richardofsussex Phase 1 of the citation generator with permanent urls (#1854) for FreeREG has been pushed to github and deployed to test3 of freereg for testing and evaluation.

All citation formats work BUT I  do not like some of the formatting. Placement of certain punctuation makes the cut and paste of the url a little unclear. I dont know if that is the decision of the orginal citation format or the way our SOC person did it

(You will note that the url contains the specific server being used. when deployed to production it will pick up the appropriate server; it has been done this way to aid in testing)

Would any comments on specific formats please state the format in question.

**Comment by Captainkirkdawson** [comment]
I have called this phase 1 as it contains some application dependent code. Phase 2 will be application independent but that is dependent on making the FR code base support multi modal control.

**Comment by richardofsussex** [comment]
No comments on the specific citation formats, but I would find it helpful if the citation appeared in a pop-up window (rather than a separate tab) with a big 'Copy' button which (when pressed) causes the citation to be copied to the clipboard. That would finesse the cut-and-paste issues you mention.

**Comment by Captainkirkdawson** [comment]
Thank you Richard. Would it be OK if your copy suggestion was dealt with subsequently? We have been trying to get something out the door for so long it would be nice to get this implementation out the door.
My concern was not the cut and paste of the citation itself but the future extraction of the url from the citation from wherever the researcher stored it ie it is important that they not include extraneous punctuation that may be in the citation.

**Comment by richardofsussex** [comment]
OK: I don't want to hold things up. Maybe that could be a new issue.

**Comment by PatReynolds** [comment]
Placement of punctuation is set by a Citation standard (e.g. MLA) so beyond our control.  However, having a pop up is a good new idea, as is having a generator / place that _just_ gives the permanent URL.  I will make new stories.

**Comment by Captainkirkdawson** [comment]
Could I urge a review of the specific text of the citations. I have no knowledge of their correctness. test3

**Comment by DeniseColbert** [comment]
@PatReynolds to check on test3

**Comment by PatReynolds** [comment]
All working fine on FreeREG.
